backend/experiment/actions/final.py

```diff
@@ -1,45 +1,174 @@
 from django.utils.translation import gettext_lazy as _
+from typing import TypedDict
 
-from experiment.serializers import serialize_social_media_config
+from experiment.serializers import serialize_social_media_config, SocialMediaConfigDto
 from session.models import Session
 
 from .base_action import BaseAction
 
 
+class ButtonDto(TypedDict):
```

Comment:
What does `Dto` stand for?

Reply:
Ah, I see, "data transfer object". Judging from extensive sampling of programmers (n=1) who have no idea what this is about, perhaps we can find other terminology? "Serialization" or "Response" perhaps?

Reply:
On a side note: while it's definitely a good idea to add types to this, I'm not sure if docstrings are really necessary here. After all, users of the interface will not directly call the functions used for serialization.

Reply:
> Ah, I see, "data transfer object". Judging from extensive sampling of programmers (n=1) who have no idea what this is about, perhaps we can find other terminology? "Serialization" or "Response" perhaps?

It's a fairly common concept in the software industry afaik, perhaps not in de python universe? I've called them `...Configuration` and `...Response` now, depending on the context. The `FinalResponse` is really a response, the other classes are both used in the constructor as well as the response.

> On a side note: while it's definitely a good idea to add types to this, I'm not sure if docstrings are really necessary here. After all, users of the interface will not directly call the functions used for serialization.

Related to what I wrote here above, the `ButtonConfiguration` and the `LogoConfiguration` types are used in the constructor of the final action, so it might make sense to show these too in the docs. What do you think?